### PR TITLE
man: document that generators can generate instances+templates and re…

### PR DESCRIPTION
**Pull request by poettering**
…gular unit files

This addition is kept brief on purpose, since in order to write a good
generator users don't really need to grok templating/instantiation.

Fixes: #7257

### CODE REVIEW

File: man/systemd.generator.xml [hunk 1]
```diff
@@ -74,17 +74,14 @@
...
-    Generators can dynamically generate unit files or create symbolic
-    links to unit files to add additional dependencies, thus extending
-    or overriding existing definitions. Their main purpose is to
-    convert configuration files that are not native unit files
-    dynamically into native unit files.</para>
+    <para>Generators are small executables that live in <filename>&systemgeneratordir;/</filename> and other
+    directories listed above.
+    <citerefentry><refentrytitle>systemd</refentrytitle><manvolnum>1</manvolnum></citerefentry> will execute those
+    binaries very early at bootup and at configuration reload time — before unit files are loaded.  Generators may
+    dynamically generate unit files (regular ones, instances as well as templates) or create symbolic links to unit
```
Comment (boucman) [inline man/systemd.generator.xml]: maybe mention drop-ins  and wants/ type diretcories too ?

That question gets asked from time to time on IRC...

Comment (poettering) [reply man/systemd.generator.xml]: true! will update

**Comment by poettering** [comment]
force pushed a new version with @boucman's point addresses

**Comment by yuwata** [comment]
lgtm.

**Pull request merged**